rust/macros/module.rs

```diff
@@ -208,7 +208,7 @@ pub(crate) fn module(ts: TokenStream) -> TokenStream {
             #[used]
             static __IS_RUST_MODULE: () = ();
```

Comment:
Why is all of this stuff not placed inside of `const _: () = { ... };`? That way it is inaccessible to the outside and we can avoid the `__` naming (this also allows multiple `module!` invocations in the same module).

We must do this, otherwise people can just call `__exit()` anywhere (since it is a safe function).

Reply:
I think it is fine for `global_asm!` to be outside. The functions used by `global_asm!` are `#[no_mangle]`, so they would not be forced out?

Reply:
I think eventually we'd want to switch to `asm_sym` and stop doing the `no_mangle` stuff.

Reply:
> The functions used by global_asm! are #[no_mangle], so they would not be forced out?

By the way, I thought `global_asm!` was mostly textual? i.e. `no_mangle` would be required anyway unless we guessed the mangled name, no? (without using `sym` parameters I mean).

Reply:
> I think eventually we'd want to switch to `asm_sym` and stop doing the `no_mangle` stuff.

That is nice, stable in 1.66, sounds good to me (for those that the C side does not need).

Reply:
And with `asm_sym`, we could create a module inside of the `const _` block where we put the `global_asm!`.

Reply:
> people can still refer to `__private_module::exit` and call the `exit` function.

How? I am talking about making the functions fully private, i.e. private-in-private, which would still export the symbol.

Reply:
that also works, but if you need them to be public, you can use `const _`. I think it is just better to make them fully inaccessible even if they require `unsafe` like the `pub extern \"C\" fn __{name}_init()` or `pub static __{name}_initcall` .
Also `const _` avoids compile errors for people who name their module the same as the macro-generated one.

Reply:
I support making things as inaccessible as possible, but if the only difference is that we avoid a compile-time error that should not be triggered, then I think simplicity is best vs. `const _: () { ... }` + a `mod` inside for `global_asm!` and/or changing the order.

I see anonymous modules were considered as an alternative for unnamed constants -- that would have been cleaner for this use case and simpler to understand.